dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DataOwnerControllerTest.kt

```diff
@@ -0,0 +1,245 @@
+            dataSet,
+            reportingPeriod,
+            false,
+        )
+    }
+
+    private fun checkErrorMessageForUnauthorizedException(clientException: ClientException) {
+        assertErrorCodeForClientException(clientException, 403)
+        val responseBody = (clientException.response as ClientError<*>).body as String
+        assertTrue(responseBody.contains("Access Denied"))
+    }
+
+    private fun checkErrorMessageForUnknownDataOwnerException(clientException: ClientException, companyId: UUID) {
+        assertErrorCodeForClientException(clientException, 404)
+        val responseBody = (clientException.response as ClientError<*>).body as String
+        assertTrue(responseBody.contains("No data owners found"))
+        assertTrue(
+            responseBody.contains(
+                "The companyId '$companyId' do not have any data owners.",
+            ),
+        )
+    }
+
+    @Test
+    fun `post data owners either to an unknown company or as a non admin and check exceptions`() {
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        val randomCompanyId = UUID.randomUUID()
+        val userId = UUID.randomUUID()
+        val postExceptionForUnknownCompany = assertThrows<ClientException> {
+            dataOwnerApi.postDataOwner(randomCompanyId, userId)
+        }
+        checkErrorMessageForUnknownCompanyException(postExceptionForUnknownCompany, randomCompanyId)
+        val headExceptionForInvalidCompany = assertThrows<ClientException> {
+            dataOwnerApi.isUserDataOwnerForCompany(randomCompanyId, userId)
+        }
+        assertErrorCodeForClientException(headExceptionForInvalidCompany, 400)
+        val companyId = UUID.fromString(
+            apiAccessor.uploadOneCompanyWithRandomIdentifier().actualStoredCompany.companyId,
+        )
+        val headExceptionForNotFoundDataOwner = assertThrows<ClientException> {
+            dataOwnerApi.isUserDataOwnerForCompany(companyId, userId)
+        }
+        assertErrorCodeForClientException(headExceptionForNotFoundDataOwner, 404)
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(
+            TechnicalUser.entries.filter { it != TechnicalUser.Admin }.random(),
+        )
+        val postExceptionForUnauthorizedRequest = assertThrows<ClientException> {
+            dataOwnerApi.postDataOwner(companyId, userId)
+        }
+        checkErrorMessageForUnauthorizedException(postExceptionForUnauthorizedRequest)
+        checkHeadException(companyId, userId)
+    }
+
+    private fun checkHeadException(companyId: UUID, userId: UUID) {
+        val headExceptionForUnauthorizedRequest = assertThrows<ClientException> {
+            dataOwnerApi.isUserDataOwnerForCompany(companyId, userId)
+        }
+        assertErrorCodeForClientException(headExceptionForUnauthorizedRequest, 403)
+    }
+
+    @Test
+    fun `delete unknown data owner from an existing and from a non existing company`() {
```

Comment:
There seem to be more than one test contained here. Maybe split it along the different test cases